app/scripts/metamask-controller.js

```diff
@@ -5355,6 +5549,56 @@ export default class MetamaskController extends EventEmitter {
     });
   }
 
+  /**
+   * Sorts a list of multichain account addresses by most recently selected by using
+   * the lastSelected value for the matching InternalAccount object stored in state.
+   *
+   * @param {string[]} [accounts] - The list of accounts addresses to sort.
+   * @returns {string[]} The sorted accounts addresses.
+   */
+  sortMultichainAccountsByLastSelected(accounts) {
+    const internalAccounts = this.accountsController.listMultichainAccounts();
+
+    return accounts.sort((firstAddress, secondAddress) => {
```

Comment:
this inner logic can be DRYed with `sortAccountsByLastSelected`.

maybe also rename `sortAccountsByLastSelected` to `sortEvmAccountsByLastSelected`?

Reply:
working on this

Reply:
https://github.com/MetaMask/metamask-extension/pull/29887/commits/500aef3e6646f51560fe4121f2e40203e904715b